### ISSUE TITLE
Extremely slow youtube downloads

### ISSUE THREAD

**Issue by theofficialgman** [subhra74/xdm#828]
**Describe the bug**
I have verified I am using the latest youtube-dl (or yt-dlp renamed to youtube-dl) inside the /opt/xdman folder.
manually running youtube-dl and downloading is FAST (as fast as my internet at 100mbps+). When using youtube-dl through XDM, the speeds is EXTREMELY SLOW (only a few kbps).

**To Reproduce**
1. update to the latest yt-dlp and name it youtube-dl in the /opt/xdman folder
2. download a youtube video using the download videos dropdown

**Expected behavior**
I expected XDM to be the same speed as using youtube-dl

**please complete the following information:**
 - OS: Linux
 - XDM Version 7.2.11

**Comment by theofficialgman** [comment]
this is due to an issue with how xdm uses youtube-dl. the links in the json can NOT be downloaded from directly with full speed
see: https://github.com/ytdl-org/youtube-dl/issues/30568#issuecomment-1123082261

**Comment by subhra74** [comment]
@theofficialgman Thanks for submitting the issue, added to the list of bugs

**Comment by nuckable** [comment]
You can probably get yt-dlp https://github.com/yt-dlp/yt-dlp copy the executable to the xdm-directory and rename it to be the same as the existing youtube-dl.
I haven't tested this myself because I don't use XDM to download youtube videos (since I use yt-dlp directly) but it should probably work just fine.

**Comment by theofficialgman** [comment]
@nuckable nope... See the thread in my above post. That does not work for xdm
